### PR TITLE
#190: re-do move explanations to selections to work with templating

### PR DESCRIPTION
**Pull request by geziefer**
I deleted the wall of text at the beginning and moved all known selection dependencies to each of the selections leaving only a short note at the top.
Each dependency was described from the point of view of the selection.

Additionally, I deleted the unused 4th column of the grid in the hope that I haven't overlooked something.

UI templating was now regarded.

### CODE REVIEW

**Comment by m-reza-rahman** [comment]
Hi @scottkurz, how does this look to you?

File: ui/src/main/webapp/index.xhtml [hunk 1]
```diff
@@ -7,65 +7,68 @@
...
-              <p:ajax event="change" listener="#{project.onRuntimeChange}"
-                      process="@form" update="@form" />
-            </p:selectOneRadio>
+            <p:column>
+              <p:selectOneRadio required="true" value="#{project.runtime}">
+                <f:selectItems value="#{project.runtimes}" />
+                <p:ajax event="change" listener="#{project.onRuntimeChange}"
+                        process="@form" update="@form" />
+              </p:selectOneRadio>
+              <h:outputText value="Tom EE requires Web profile, Tom EE and Open Liberty require Jakarta EE below 10, GlassFish requires no docker suport"/>
```
Comment (scottkurz) [inline ui/src/main/webapp/index.xhtml:58]:
```suggestion
              <h:outputText value="TomEE requires Web profile, TomEE and Open Liberty require Jakarta EE below 10, GlassFish requires no Docker support"/>
```

File: ui/src/main/webapp/index.xhtml [hunk 2]
```diff
@@ -7,65 +7,68 @@
...
-              <p:ajax event="change" listener="#{project.onDockerChange}"
-                      process="@form" update="@form" />
-            </p:selectOneRadio>
+            <p:column>
+              <p:selectOneRadio required="true" value="#{project.docker}">
+                <f:selectItems value="#{project.dockerFlags}" />
+                <p:ajax event="change" listener="#{project.onDockerChange}"
+                        process="@form" update="@form" />
+              </p:selectOneRadio>
+              <h:outputText value="Docker support yes requires runtime other than GlassFish"/>
```
Comment (scottkurz) [inline ui/src/main/webapp/index.xhtml:69]:
```suggestion
              <h:outputText value="Docker support requires a runtime other than GlassFish"/>
```

**Review by scottkurz — CHANGES_REQUESTED** [review]
Yeah I think that's a nicer look (to move the comments inline next to the various options).
A couple minor spelling/capitalization edits:

**Pull request merged**